### DISCUSSION TITLE
Is there a way to refresh / invalidate the Ferrum node ID?

### DISCUSSION THREAD

**Discussion by shepmaster** [rubycdp/cuprite#241] [Q&A]
I was trying out Cuprite and it works across almost all of our tests... except one. In that test, it fails with `Capybara::Cuprite::ObsoleteNode`:

```
Capybara::Cuprite::ObsoleteNode Exception: The element you are trying to interact with is either not part of the DOM, or is not currently visible on the page (perhaps display: none is set). It is possible the element has been replaced by another element and you meant to interact with the new element. If so you need to do a new find in order to get a reference to the new element.
```

A problem with the suggestion is that I'm not doing a `find` directly, but have the rough structure of:

```rb
within('.jobs') do
  expect(page).to have_elements('.jobs__name').with_values(['A job', 'Z job'])
end
```

`have_elements` is a custom RSpec matcher that essentially does:

```rb
page.document.synchronize(Capybara.default_max_wait_time, errors: errors(page)) do
  page.all(css_selector, count: @expected_values.length)
  # ...
end
```

It's this `page.all` that triggers the error, as (presumably) the element for the `.jobs` class is no longer valid. 

Ideally, a node reporting `ObsoleteNode` would trigger re-fetching the node's ID.

**Comment by route** [comment]
I believe it works like you described, but eventually timeout is triggered and `ObsoleteNode` returned.

**Reply by shepmaster** [reply]
Ok, in that case I'll open [an issue with reproduction steps](https://github.com/rubycdp/cuprite/issues/242).
